### PR TITLE
feat: JOJ3 issue enhancement

### PR DESCRIPTION
**Pull request by mQzLjP**
Some improvement to JOJ3 issues in this PR:
- Removed the timestamp. (m says it's unnecessary)
- Add exercise names to the issue title, integrated in CLI (might need further integration to meet with JOJ3 repo)
- Make details (namely comments) collapsible
- Add make-up comments if there's compile error causing a 0, or if the code passed

### CODE REVIEW

File: joint_teapot/utils/joj3.py [hunk 1]
```diff
@@ -201,13 +201,22 @@ def generate_title_and_comment(
         comment += "\n"
         for i, result in enumerate(stage["results"]):
             if not single_case:
-                comment += f"### Case {i} - Score: {result['score']}\n"
+                comment += (
+                    f"<details><summary>Case {i} - Score: {result['score']}</summary>\n"
+                )
             if result["comment"].strip() != "":
                 comment += f"{result['comment']}\n"
+            else:
```
Comment (BoYanZh) [inline joint_teapot/utils/joj3.py:209]: We will never have this field for failed or passed in JOJ3. No need to comment failed or passed here, the parser will generate it if needed.

File: joint_teapot/utils/joj3.py [hunk 2]
```diff
@@ -201,13 +201,22 @@ def generate_title_and_comment(
         comment += "\n"
         for i, result in enumerate(stage["results"]):
             if not single_case:
-                comment += f"### Case {i} - Score: {result['score']}\n"
+                comment += (
+                    f"<details><summary>Case {i} - Score: {result['score']}</summary>\n"
+                )
             if result["comment"].strip() != "":
                 comment += f"{result['comment']}\n"
```
Comment (BoYanZh) [inline joint_teapot/utils/joj3.py:208]: Let us collapse the comment on a single case for consistency, as it might be long if a single case diff is failed.

**Review by BoYanZh — CHANGES_REQUESTED** [review]

File: joint_teapot/utils/joj3.py [hunk 3]
```diff
@@ -201,13 +201,22 @@ def generate_title_and_comment(
...
+                    comment += "Failed...\n"
+                else:
+                    comment += "Passed!\n"  # FIXME: This is a temporary solution. The comment should be better generated directly in the json file.
             total_score += result["score"]
+            if not single_case:
+                comment += "</details>\n\n"
         comment += "\n"
-    now = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
-    title = f"JOJ3 Result {now} - Score: {total_score}"
+    # now = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
```
Comment (BoYanZh) [inline joint_teapot/utils/joj3.py:218]: Remove this line if not needed any more.

**Comment by mQzLjP** [comment]
@BoYanZh Fixed! For the parser generating failed/passed, it seems m want it desperately :joy:

File: joint_teapot/utils/joj3.py [hunk 4]
```diff
@@ -82,9 +82,9 @@ def generate_scoreboard(
     submitter_row[columns.index("total")] = str(total)
 
     now = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
-    submitter_row[
-        columns.index("last_edit")
-    ] = now  # FIXME: Delete this in formal version
+    submitter_row[columns.index("last_edit")] = (
+        now  # FIXME: Delete this in formal version
```
Comment (BoYanZh) [inline joint_teapot/utils/joj3.py:86]: Have you installed `pre-commit` and run it? I think the original indent is correct.

File: joint_teapot/utils/joj3.py [hunk 5]
```diff
@@ -201,13 +201,17 @@ def generate_title_and_comment(
         comment += "\n"
         for i, result in enumerate(stage["results"]):
             if not single_case:
-                comment += f"### Case {i} - Score: {result['score']}\n"
+                comment += (
+                    f"<details><summary>Case {i} - Score: {result['score']}</summary>\n"
+                )
+            else:
+                comment += "<details><summary>Details</summary>\n"
```
Comment (BoYanZh) [inline joint_teapot/utils/joj3.py:208]: What about showing the score as the summary and removing line 199 and 200?

File: joint_teapot/utils/joj3.py [hunk 6]
```diff
@@ -179,7 +179,7 @@ def generate_failed_table(
 
 
 def generate_title_and_comment(
-    score_file_path: str, action_link: str, run_number: str
+    score_file_path: str, action_link: str, run_number: str, exercise_name: str
```
Comment (BoYanZh) [inline joint_teapot/utils/joj3.py:182]: This `exercise_name` should also be added to other 2 joj3 commands. And it should use the comment of metadata stage (line 58) as a fallback if exercise name is Unknown. Just for a smoother migration.

**Review by BoYanZh — CHANGES_REQUESTED** [review]

File: joint_teapot/utils/joj3.py (hunk 5 above)
Comment (mQzLjP) [reply joint_teapot/utils/joj3.py:208]: For consistency, maybe I'll just remove the single_case variable?

File: joint_teapot/utils/joj3.py (hunk 6 above)
Comment (mQzLjP) [reply joint_teapot/utils/joj3.py:182]: How to have access to the data written in metadata stage? I don't have access to the scoreboard, so I'm not quite sure

File: joint_teapot/utils/joj3.py (hunk 4 above)
Comment (mQzLjP) [reply joint_teapot/utils/joj3.py:86]: It seems cz in my lazygit will skip pre-commit.. sorry

Comment (mQzLjP) [reply joint_teapot/utils/joj3.py]: fixed

File: joint_teapot/utils/joj3.py (hunk 6 above)
Comment (mQzLjP) [reply joint_teapot/utils/joj3.py:182]: Changed the code to let the user set the exercise_name at metadata procedure.

**Review by BoYanZh — APPROVED** [review]

**Pull request merged**